driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -378,6 +406,27 @@ private boolean addConnectionIfUnderMaximum() {
         }
     }
 
+    private PooledConnection tryResurrectFromTrash() {
+        long highestMaxIdleTime = System.currentTimeMillis();
+        PooledConnection chosen = null;
+
+        while (true) {
+            for (PooledConnection connection : trash)
+                if (connection.maxIdleTime > highestMaxIdleTime && connection.maxAvailableStreams() > MIN_AVAILABLE_STREAMS) {
```

Comment:
Was just thinking that this could be made more picky and also not select connections that are at the max number of requests on that connection per the pooling options.